core/pom.xml

```diff
@@ -83,6 +83,11 @@
             <groupId>org.mule.runtime</groupId>
             <artifactId>mule-profiling-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.mule.runtime</groupId>
+            <artifactId>mule-module-feature-management</artifactId>
```

Comment:
is this actually needed in core? cannot the mechanism be coupled to just an interface?

Reply:
Created ticket MULE-19954.
The idea is to make feature flagging fully decoupled from the core module.
It should be injectable as a service through an interface. Possibly the managed service should be at the container level.
This is currently built and added to the TransientRegistry manually when the muleContext is created.